app/models/user.rb

```diff
@@ -3,12 +3,17 @@ class User < ApplicationRecord
 
   has_many :category_suggestions
 
+  has_one_attached :avatar
+
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable, and :omniauthable
   devise :database_authenticatable, :recoverable, :rememberable, :trackable, :validatable
 
+  before_create :assign_avatar
```

Comment:
Out of curiosity, why do we assign an avatar before creation? Doesn't that risk an orphaned avatar if the user isn't successfully created?

Reply:
Hrm, good point. After_create is probably better.